File: 1. Arrays & Strings/3_anagrams.java
import java.util.HashMap;

class Source {

    public static HashMap<Character, Integer> convertToMap(String s) {
        HashMap<Character, Integer> map = new HashMap<>();

        for (char c : s.toCharArray()) {
            if (map.get(c) == null) {
                map.put(c, 1); // new char set to 1 occ
            } else { // same char, add occ
                int current = map.get(c); // get current char occ
                map.put(c, current + 1); // overwrite + 1 occ
            }
        }

        return map;
    }

    public static boolean anagrams(String s1, String s2) {
        boolean result = convertToMap(s1).equals(convertToMap(s2));
        return result;
    }
}

File: 1. Arrays & Strings/3_anagrams.py
def convert_to_map(s):
    char_map = {}

    for char in s:
        if char not in char_map:
            char_map[char] = 1  # new char set to 1 occ
        else:
            char_map[char] += 1  # same char, add occ
    
    return char_map

def anagrams(s1, s2):
    return convert_to_map(s1) == convert_to_map(s2)
